src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -587,4 +587,26 @@ public static void SetPowerShellAssemblyLoadContext([MarshalAs(UnmanagedType.LPW
             PowerShellAssemblyLoadContext.InitializeSingleton(basePaths);
         }
     }
+
+    /// <summary>
+    /// Provides helper functions to faciliate calling managed code from a native PowerShell host.
+    /// </summary>
+    public static unsafe class PowerShellUnsafeAssemblyLoad
+    {
+        /// <summary>
+        /// Load an assembly in memory from unmanaged code.
+        /// </summary>
+        /// <param name="data">
+        /// Unmanaged pointer to assembly data buffer.
+        /// </param>
+        /// <param name="size">
+        /// Size in bytes of the assembly data buffer.
+        /// </param>
+        [UnmanagedCallersOnly]
+        public static void LoadAssemblyFromNativeMemory(IntPtr data, int size)
```

Comment:
The use of this API should be logged, at minimum on Windows platform.
@TravisEz13  Do you have any ideas on where it should be logged?  I am thinking a new ETW PowerShell log entry for this.

Reply:
After additional discussion with @PaulHigin, we agree logging can be added in a separate PR. Issue #15293 was filed to track that work.